pkg/contract/contract.go

```diff
@@ -319,15 +327,25 @@ func TxToMethod(
 	}
 	defer client.Close()
 	contract := bind.NewBoundContract(contractAddress, *abi, client, client, client)
-	txOpts, err := evm.GetTxOptsWithSigner(client, privateKey)
-	if err != nil {
-		return nil, nil, err
+	var txOpts *bind.TransactOpts
+	if generateRawTxOnly {
+		txOpts = &bind.TransactOpts{
+			From:   from,
```

Comment:
why can't we just use  crypto.PubkeyToAddress(pk.PublicKey), this way no need to use from in argument